scm/build/Dockerfile

```diff
@@ -1,24 +1,139 @@
  && mkdir -p ${GOPATH}/bin \
  && mkdir -p ${GOPATH}/src \
- && curl -fsSL -k https://dl.google.com/go/go${GO_VERSION}.src.tar.gz | tar zx --strip-components=1 -C ${GOPATH} \
- && cd ${GOPATH}/src/ \
+ && curl -fsSL -k https://dl.google.com/go/go${GO_VERSION}.src.tar.gz | tar zx --strip-components=1 -C ${GOROOT} \
+ && cd ${GOROOT}/src/ \
  && ./make.bash \
  && rm -rf /var/cache/apk/* \
  && rm -rf /var/lib/apt/lists/* \
  && rm -rf /tmp/*
 
-ENTRYPOINT /build.sh
+RUN mkdir -p ${GOPATH}/src/github.com/nokia/danm
+COPY glide.yaml glide.lock ${GOPATH}/src/github.com/nokia/danm/
+COPY scm/build/build-deps.sh /
+
+WORKDIR ${GOPATH}/src/github.com/nokia/danm
+
+RUN /build-deps.sh
+
+
+#
+# Stage: Build actual binaries. This builder is used to build all DANM
+# related binaries.
+#
+# It is separated from the "build-deps" stage because mostly because
+# building the dependencies takes quite a lot of time and they (presumably)
+# don't change often -- so this could be factored out into a reusable
+# base image.
+#
+FROM build-deps AS builder
+
+COPY / ${GOPATH}/src/github.com/nokia/danm/
+
+ARG COMMIT_HASH
+ARG LATEST_TAG
+
+RUN scm/build/build.sh
+
+
+#
+# Stage: Netwatcher
+#
+FROM alpine:latest AS netwatcher
+MAINTAINER Levente Kale <levente.kale@nokia.com>
+
+COPY --from=builder /go/bin/netwatcher /usr/local/bin/netwatcher
+
+RUN apk add --no-cache --virtual .tools curl libcap iputils  \
+ && adduser -u 147 -D -H -s /sbin/nologin danm \
+ && chown root:danm /usr/local/bin/netwatcher \
+ && chmod 750 /usr/local/bin/netwatcher \
+ && setcap cap_sys_ptrace,cap_sys_admin,cap_net_admin=eip /usr/local/bin/netwatcher \
+ && setcap cap_net_raw=eip /usr/sbin/arping \
+ && apk del .tools
+
+USER danm
+
+WORKDIR /
+ENTRYPOINT ["/usr/local/bin/netwatcher"]
+
+
+
+#
+# Stage: Svcwatcher
+#
+FROM alpine:latest AS svcwatcher
+MAINTAINER Levente Kale <levente.kale@nokia.com>
+
+COPY --from=builder /go/bin/svcwatcher /usr/local/bin/svcwatcher
+
+RUN apk add --no-cache --virtual .tools curl libcap iputils  \
+ && adduser -u 147 -D -H -s /sbin/nologin danm \
+ && chown root:danm /usr/local/bin/svcwatcher \
+ && chmod 750 /usr/local/bin/svcwatcher \
+ && apk del .tools
+
+USER danm
+
+WORKDIR /
+ENTRYPOINT ["/usr/local/bin/svcwatcher"]
+
+
+
+#
+# Stage: Webhook
+#
+FROM alpine:latest AS webhook
+MAINTAINER Levente Kale <levente.kale@nokia.com>
+
+COPY --from=builder /go/bin/webhook /usr/local/bin/webhook
+
+RUN apk add --no-cache --virtual .tools curl libcap iputils  \
+ && adduser -u 147 -D -H -s /sbin/nologin danm \
+ && chown root:danm /usr/local/bin/webhook \
+ && chmod 750 /usr/local/bin/webhook \
+ && apk del .tools
+
+USER danm
+
+WORKDIR /
+
+ENTRYPOINT ["/usr/local/bin/webhook"]
+
+
+
+#
+# Stage: CNI plugins daemonset
+#
+FROM alpine:latest AS danm-cni-plugins
+
+VOLUME ["/host/cni"]
+
+RUN mkdir /cni
+
+COPY --from=builder /go/bin/danm /cni/danm
+COPY --from=builder /go/bin/fakeipam /cni/fakeipam
+COPY --from=builder /go/bin/macvlan /cni/macvlan
```

Comment:
These can be collapsed in one line like this:
`COPY --from=builder --chown=user:group /go/bin/danm /go/bin/fakeipam /go/bin/macvlan /cni/`

Reply:
Done.